apps/api/src/queries/photograph/photograph-queries-fetch-by-id.e2e.spec.ts

```diff
@@ -0,0 +1,538 @@
+                                );
+
+                                expect(res.status).toBe(httpStatusCodes.ok);
+
+                                // We don't expose actions to non-admin users
+                                expect(res.body.actions).toEqual([]);
+                            });
+                        });
+                    });
+
+                    describe(`when the term is not published and the user does not have access`, () => {
+                        beforeEach(async () => {
+                            await photographQueryRepository.createMany([
+                                clonePlainObjectWithOverrides(targetPhotographView, {
+                                    isPublished: false,
+                                    // no special access here
+                                    accessControlList: new AccessControlList(),
+                                }),
+                            ]);
+                        });
+
+                        // We pretend the resource does not exist when the user
+                        // does not have access to this term
+                        it(`should return not found (404)`, async () => {
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(photographId)
+                            );
+
+                            expect(res.status).toBe(httpStatusCodes.notFound);
+                        });
+                    });
+
+                    describe(`when the term is not published but the user has access`, () => {
+                        beforeEach(async () => {
+                            await photographQueryRepository.createMany([
+                                privatePhotographThatUserCanAccess,
+                            ]);
+                        });
+
+                        it(`should return the expected result`, async () => {
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(photographId)
+                            );
+
+                            expect(res.status).toBe(httpStatusCodes.ok);
+
+                            // We don't expose actions to non-admin users
+                            expect(res.body.actions).toEqual([]);
+                        });
+                    });
+
+                    describe(`when there is no photograph with the given Id`, () => {
+                        it(`should return not found (404)`, async () => {
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(buildDummyUuid(458))
+                            );
+
+                            expect(res.status).toBe(httpStatusCodes.notFound);
+                        });
+                    });
+                });
+            });
+        });
+    });
+});
```

Comment:
I wonder if there's something we can do to make this test less tedious to write. I see lots of copy and paste errors which is a sign that it's too much.

The big issue is that we have combinatorial complexity. There's a combination of 
- User (public, authenticated ordinary user, admin, coscrad)
- Resource State (public, private, private with ACL access for given user)

I am not a fan of using test case builder patterns that hide Jest's describe \ it \ expect. The place to build helpers is for:
- test setup
- common assertions

It would be great if we could put some thought to this as I think it would improve our velocity and the readability of all query tests if we could sort this out now.